git-link.el

```diff
@@ -666,6 +666,54 @@ return (FILENAME . REVISION) otherwise nil."
     (git-link--remote)))
 
 ;;;###autoload
+(defun git-link-url (file-or-buffer remote &optional start end use-branch-p)
```

Comment:
Some thoughts/issues here:

1. I think we should default `remote` to `git-link-default-remote`. 
1. If the current buffer is in directory `X` but this is called like `(git-link-url "/path/to/Y/foo.txt" "origin" 10)`, where `Y` is a different repository root, it should return a link to `Y/foo.txt`'s repo but now it links to something random in repo `X`. 
1. What to do with invalid line numbers, e.g., when `end` is after or equal to start, when `start` is `nil` and `end` is given? Thinking we check for these cases and signal an error.

Reply:
> ... but item 2 strikes me as out of scope for this PR.

When calling it as a separate function it cannot depend on state of some buffer. For example, if one creates an emacs package that calls `git-link-url` the call will depend on the state of some git repo outside the function call so bugs would occur. Make sense? 

> I agree with items 1 and 3, 

Maybe 2/3 is fine? Do they togeather outweigh the bug? 

> Do you see another way of achieving 2 that only impacts git-link-url?

I think this is part of the reason why this function has taken so long to exist. If I remember correctly you have to change a bit more than meets the eye. It has been since Dec or so since I had a look at the code. Will have to wait to look again. But if you think you gotz a solution then add it. But if you are concerned about your time then maybe wait 'till I can have an in-depth look again.

Reply:
Also test coverage not that good. [Exhibit A](https://github.com/sshaw/git-link/issues/95).